api/metadata/metadata.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/constants"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const versionKey = "version"
+
+var (
+	versionValue = constants.Version
+	disabled     = false
+)
+
+// SetVersion is used only for tests to set the version value that the client
+// will send in the GRPC metadata.
+func SetVersion(version string) {
+	versionValue = version
```

Comment:
I would consider protecting the global variables by a mutex or ensure atomicity some other way to account for potential concurrent access.

Reply:
Good point, added these globals as kind of a hack for testing so didn't really think about it.  I've wrapped access to these in a RWMutex in the latest commit.